util/featureflag/featureflag.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 PingCAP, Inc. Licensed under Apache-2.0.
+
+package featureflag
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/gin-gonic/gin"
+
+	"github.com/pingcap/tidb-dashboard/util/rest"
+)
+
+var ErrFeatureUnsupported = rest.ErrForbidden.NewSubtype("feature_unsupported")
+
+type FeatureFlag struct {
+	name        string
+	constraints []string
+	isSupported bool
+}
+
+func newFeatureFlag(name, targetVersion string, constraints ...string) *FeatureFlag {
+	f := &FeatureFlag{name: name, constraints: constraints}
+	f.isSupported = f.isSupportedIn(targetVersion)
+	return f
+}
+
+func (f *FeatureFlag) Name() string {
+	return f.name
+}
+
+func (f *FeatureFlag) IsSupported() bool {
+	return f.isSupported
+}
+
+// VersionGuard returns gin.HandlerFunc as guard middleware.
+// It will determine if features are available in the target version.
+func (f *FeatureFlag) VersionGuard() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if !f.isSupported {
+			_ = c.Error(ErrFeatureUnsupported.New(f.name))
+			c.AbortWithStatus(http.StatusForbidden)
```

Comment:
This may be not compatible with the rest.ErrorHandlerFn. Please add a test to verify it.